ldap/servers/slapd/back-ldbm/ldbm_search.c

```diff
@@ -1538,7 +1538,7 @@ ldbm_back_next_search_entry(Slapi_PBlock *pb)
             goto bail;
         }
         /* check lookthrough limit */
-        if (llimit != -1 && sr->sr_lookthroughcount >= llimit) {
+        if (llimit != -1 && sr->sr_current >= llimit) {
```

Comment:
We don't need to test this in the while loop, since sr_current is the number of ids in the partial candidate set (ie it's never going to change within the loop). So we actually should move this up to line ~1514.

Reply:
Hi @Firstyear 

As we iterate through the candidate list on +1573,  sr_current is incremented as we get the id for each entry. So we need to keep the check for lookthroughlimit in the while loop.

Reply:
Ahh yes, you're correct. I had to re-read the code a few times to be sure. You could actually use:

```
if (llimit != 1 && sr->sr_candidates->b_nids >= llimit) {
}
```

But again, outside the loop. I'd want @progier389 to sanity check though too since this changes the approach a bit.